pkg/querier/querier.go

```diff
@@ -72,9 +72,9 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet, logger log.Logger) {
 	cfg.StoreGatewayClient.RegisterFlagsWithPrefix("querier.store-gateway-client", f)
 	f.BoolVar(&cfg.Iterators, "querier.iterators", false, "Use iterators to execute query, as opposed to fully materialising the series in memory.")
 	f.BoolVar(&cfg.BatchIterators, "querier.batch-iterators", true, "Use batch iterators to execute query, as opposed to fully materialising the series in memory.  Takes precedent over the -querier.iterators flag.")
-	f.DurationVar(&cfg.QueryIngestersWithin, queryIngestersWithinFlag, 13*time.Hour, "Maximum lookback beyond which queries are not sent to ingester. 0 means all queries are sent to ingester.")
+	f.DurationVar(&cfg.QueryIngestersWithin, queryIngestersWithinFlag, 13*time.Hour, "Maximum lookback beyond which queries are not sent to ingester. 0 means all queries are sent to ingester. Should be at least an hour greater than -querier.query-store-after.")
 	f.DurationVar(&cfg.MaxQueryIntoFuture, "querier.max-query-into-future", 10*time.Minute, "Maximum duration into the future you can query. 0 to disable.")
-	f.DurationVar(&cfg.QueryStoreAfter, queryStoreAfterFlag, 12*time.Hour, "The time after which a metric should be queried from storage and not just ingesters. 0 means all queries are sent to store. If this option is enabled, the time range of the query sent to the store-gateway will be manipulated to ensure the query end is not more recent than 'now - query-store-after'.")
+	f.DurationVar(&cfg.QueryStoreAfter, queryStoreAfterFlag, 12*time.Hour, "The time after which a metric should be queried from storage and not just ingesters. 0 means all queries are sent to store. If this option is enabled, the time range of the query sent to the store-gateway will be manipulated to ensure the query end is not more recent than 'now - query-store-after'. Should be greater than -blocks-storage.bucket-store.ignore-blocks-within and at least an hour smaller than -querier.query-ingesters-within.")
```

Comment:
> at least an hour smaller than -querier.query-ingesters-within

Same edit here.

Reply:
I just want to save your and my time for next time when I forget how to set these.

Is there a good use case for writing a guide under https://grafana.com/docs/mimir/latest/operators-guide/configuring/ that can explain this and be useful? 

My use case was: reduce the time it takes to hit the cache so that I can test it, but that's not going to be a valid use-case for the generic user.

Reply:
> I just want to save your and my time for next time when I forget how to set these.
> 
> Is there a good use case for writing a guide under https://grafana.com/docs/mimir/latest/operators-guide/configuring/ that can explain this and be useful?
> 
> My use case was: reduce the time it takes to hit the cache so that I can test it, but that's not going to be a valid use-case for the generic user.

Write first for the user.

Reply:
You can use a phrase such as, "As a contributor to ..., ..." in the core documentation to qualify the reader.